prov/gni/test/cntr.c

```diff
@@ -797,3 +797,154 @@ Test(cntr_mt, write_wait)
 
 	dbg_printf("done\n");
 }
+
+void *do_add_cntr_mt(void *arg)
+{
+	int i = 0, ret, iters = ((int *)arg)[0];
+
+	for (; i < iters; i++) {
+		ret = fi_cntr_add(ep_write_cntrs[0], 1);
+		cr_assert(ret == FI_SUCCESS);
+	}
+
+	return NULL;
+}
+
+void *do_add_err_cntr_mt(void *arg)
+{
+	int i = 0, ret, iters = ((int *)arg)[0];
+
+	for (; i < iters; i++) {
+		ret = fi_cntr_adderr(ep_write_cntrs[0], 1);
+		cr_assert(ret == FI_SUCCESS);
+	}
+
+	return NULL;
+}
+
+Test(cntr_mt, set_add_read_cntr)
+{
+	int iters = 128, nthreads = 4, i, ret;
+	uint64_t cntr_val;
+	pthread_t threads[nthreads];
+	void *pt_ret;
+
+
+	ret = fi_cntr_set(ep_write_cntrs[0], 0);
+	cr_assert(ret == FI_SUCCESS);
+
+	cntr_val = fi_cntr_read(ep_write_cntrs[0]);
+	cr_assert_eq(cntr_val, 0, "write error counter is incorrect.");
+
+	// Create & Spawn threads
+	for (i = 0; i < nthreads; i++) {
+		cr_assert(!pthread_create(threads + i, NULL, do_add_cntr_mt,
+					  (void *) &iters));
+	}
+
+	// Wait until all threads are done
+	for (i = 0; i < nthreads; i++) {
+		cr_assert(!pthread_join(threads[i], &pt_ret));
+		cr_assert(!pt_ret);
+	}
+
+	cntr_val = fi_cntr_read(ep_write_cntrs[0]);
+	cr_assert_eq(cntr_val, (iters * nthreads), "write error counter "
+		     "is incorrect.");
+}
+
+Test(cntr_mt, set_add_read_err_cntr)
+{
+	int iters = 128, nthreads = 4, i, ret;
+	uint64_t err_cntr_val;
+	pthread_t threads[nthreads];
+	void *pt_ret;
+
+
+	ret = fi_cntr_seterr(ep_write_cntrs[0], 0);
+	cr_assert(ret == FI_SUCCESS);
+
+	err_cntr_val = fi_cntr_readerr(ep_write_cntrs[0]);
+	cr_assert_eq(err_cntr_val, 0, "write error counter is incorrect.");
+
+	// Create & Spawn threads
+	for (i = 0; i < nthreads; i++) {
+		cr_assert(!pthread_create(threads + i, NULL, do_add_err_cntr_mt,
+					  (void *) &iters));
+	}
+
+	// Wait until all threads are done
+	for (i = 0; i < nthreads; i++) {
+		cr_assert(!pthread_join(threads[i], &pt_ret));
+		cr_assert(!pt_ret);
+	}
+
+	err_cntr_val = fi_cntr_readerr(ep_write_cntrs[0]);
+	cr_assert_eq(err_cntr_val, (iters * nthreads), "write error counter "
+		     "is incorrect.");
+}
+
+void do_add_cntr_st(void *arg)
+{
```

Comment:
I'd like these functions to just be removed. For a single threaded test, these functions just add a stack where one doesn't need to exist. The code contained in these functions isn't reused anywhere, so it doesn't really serve a purpose.